### DISCUSSION TITLE
Backup failed because The dump process failed with exitcode 2 : Misuse of shell builtins : mysqldump: Couldn't execute 'SELECT COLUMN_NAME,

### DISCUSSION THREAD

**Discussion by codewithali** [spatie/laravel-backup#1303] [Q&A]
Backup failed because The dump process failed with exitcode 2 : Misuse of shell builtins : mysqldump: Couldn't execute 'SELECT COLUMN_NAME,                       JSON_EXTRACT(HISTOGRAM, '$."number-of-buckets-specified"')                FROM information_schema.COLUMN_STATISTICS                WHERE SCHEMA_NAME = 'db_main' AND TABLE_NAME = 'additional_detail_lot';': Unknown table 'COLUMN_STATISTICS' in information_schema (1109)
.
#0 /home/ashlar/laravel/vendor/spatie/db-dumper/src/DbDumper.php(263): Spatie\DbDumper\Exceptions\DumpFailed::processDidNotEndSuccessfully()
#1 /home/ashlar/laravel/vendor/spatie/db-dumper/src/Databases/MySql.php(203): Spatie\DbDumper\DbDumper->checkIfDumpWasSuccessFul()
#2 /home/ashlar/laravel/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php(247): Spatie\DbDumper\Databases\MySql->dumpToFile()
#3 [internal function]: Spatie\Backup\Tasks\Backup\BackupJob->Spatie\Backup\Tasks\Backup\{closure}()
#4 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Support/Collection.php(1034): array_map()
#5 /home/ashlar/laravel/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php(250): Illuminate\Support\Collection->map()
#6 /home/ashlar/laravel/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php(167): Spatie\Backup\Tasks\Backup\BackupJob->dumpDatabases()
#7 /home/ashlar/laravel/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php(143): Spatie\Backup\Tasks\Backup\BackupJob->createBackupManifest()
#8 /home/ashlar/laravel/vendor/spatie/laravel-backup/src/Commands/BackupCommand.php(52): Spatie\Backup\Tasks\Backup\BackupJob->run()
#9 [internal function]: Spatie\Backup\Commands\BackupCommand->handle()
#10 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array()
#11 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#12 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod()
#13 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Container/Container.php(572): Illuminate\Container\BoundMethod::call()
#14 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Console/Command.php(183): Illuminate\Container\Container->call()
#15 /home/ashlar/laravel/vendor/symfony/console/Command/Command.php(255): Illuminate\Console\Command->execute()
#16 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Console/Command.php(170): Symfony\Component\Console\Command\Command->run()
#17 /home/ashlar/laravel/vendor/spatie/laravel-backup/src/Commands/BaseCommand.php(16): Illuminate\Console\Command->run()
#18 /home/ashlar/laravel/vendor/symfony/console/Application.php(1009): Spatie\Backup\Commands\BaseCommand->run()
#19 /home/ashlar/laravel/vendor/symfony/console/Application.php(273): Symfony\Component\Console\Application->doRunCommand()
#20 /home/ashlar/laravel/vendor/symfony/console/Application.php(149): Symfony\Component\Console\Application->doRun()
#21 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Console/Application.php(89): Symfony\Component\Console\Application->run()
#22 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(122): Illuminate\Console\Application->run()
#23 /home/ashlar/laravel/artisan(37): Illuminate\Foundation\Console\Kernel->handle()
#24 {main}
Backup failed because: The dump process failed with exitcode 2 : Misuse of shell builtins : mysqldump: Couldn't execute 'SELECT COLUMN_NAME,                       JSON_EXTRACT(HISTOGRAM, '$."number-of-buckets-specified"')                FROM information_schema.COLUMN_STATISTICS                WHERE SCHEMA_NAME = 'db_main' AND TABLE_NAME = 'additional_detail_lot';': Unknown table 'COLUMN_STATISTICS' in information_schema (1109)

**Comment by codewithali** [comment]
Backup failed because The dump process failed with exitcode 2 : Misuse of shell builtins : mysqldump: Couldn't execute 'SELECT COLUMN_NAME,                       JSON_EXTRACT(HISTOGRAM, '$."number-of-buckets-specified"')                FROM information_schema.COLUMN_STATISTICS                WHERE SCHEMA_NAME = 'db_main' AND TABLE_NAME = 'additional_detail_lot';': Unknown table 'COLUMN_STATISTICS' in information_schema (1109)
.
#0 /home/ashlar/laravel/vendor/spatie/db-dumper/src/DbDumper.php(263): Spatie\DbDumper\Exceptions\DumpFailed::processDidNotEndSuccessfully()
#1 /home/ashlar/laravel/vendor/spatie/db-dumper/src/Databases/MySql.php(203): Spatie\DbDumper\DbDumper->checkIfDumpWasSuccessFul()
#2 /home/ashlar/laravel/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php(247): Spatie\DbDumper\Databases\MySql->dumpToFile()
#3 [internal function]: Spatie\Backup\Tasks\Backup\BackupJob->Spatie\Backup\Tasks\Backup\{closure}()
#4 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Support/Collection.php(1034): array_map()
#5 /home/ashlar/laravel/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php(250): Illuminate\Support\Collection->map()
#6 /home/ashlar/laravel/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php(167): Spatie\Backup\Tasks\Backup\BackupJob->dumpDatabases()
#7 /home/ashlar/laravel/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php(143): Spatie\Backup\Tasks\Backup\BackupJob->createBackupManifest()
#8 /home/ashlar/laravel/vendor/spatie/laravel-backup/src/Commands/BackupCommand.php(52): Spatie\Backup\Tasks\Backup\BackupJob->run()
#9 [internal function]: Spatie\Backup\Commands\BackupCommand->handle()
#10 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array()
#11 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#12 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod()
#13 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Container/Container.php(572): Illuminate\Container\BoundMethod::call()
#14 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Console/Command.php(183): Illuminate\Container\Container->call()
#15 /home/ashlar/laravel/vendor/symfony/console/Command/Command.php(255): Illuminate\Console\Command->execute()
#16 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Console/Command.php(170): Symfony\Component\Console\Command\Command->run()
#17 /home/ashlar/laravel/vendor/spatie/laravel-backup/src/Commands/BaseCommand.php(16): Illuminate\Console\Command->run()
#18 /home/ashlar/laravel/vendor/symfony/console/Application.php(1009): Spatie\Backup\Commands\BaseCommand->run()
#19 /home/ashlar/laravel/vendor/symfony/console/Application.php(273): Symfony\Component\Console\Application->doRunCommand()
#20 /home/ashlar/laravel/vendor/symfony/console/Application.php(149): Symfony\Component\Console\Application->doRun()
#21 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Console/Application.php(89): Symfony\Component\Console\Application->run()
#22 /home/ashlar/laravel/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(122): Illuminate\Console\Application->run()
#23 /home/ashlar/laravel/artisan(37): Illuminate\Foundation\Console\Kernel->handle()
#24 {main}
Backup failed because: The dump process failed with exitcode 2 : Misuse of shell builtins : mysqldump: Couldn't execute 'SELECT COLUMN_NAME,                       JSON_EXTRACT(HISTOGRAM, '$."number-of-buckets-specified"')                FROM information_schema.COLUMN_STATISTICS                WHERE SCHEMA_NAME = 'db_main' AND TABLE_NAME = 'additional_detail_lot';': Unknown table 'COLUMN_STATISTICS' in information_schema (1109)

**Comment by jorenvh** [comment]
I stumbled on the same issue. To fix this globally, you can add the following line your `mysqldump.cnf` file.

```
[mysqldump]
...
column-statistics=0
...
```

Resources:
* https://serverfault.com/questions/912162/mysqldump-throws-unknown-table-column-statistics-in-information-schema-1109
* https://dev.mysql.com/doc/refman/8.0/en/mysqldump.html#option_mysqldump_column-statistics

**Comment by babul** [comment]
This occurred for me with a mariadb database backend when having mysql-client installed.

I removed mysql-client and installed mariadb-client and problem resolved!
